### PR TITLE
Setup and Configure Server, Authentication, and User Management

### PR DESCRIPTION
**Pull request by micaiah-effiong**
## Summary of Changes:

- Server Setup & Configuration:
  - Added basic instructions to the README for setting up the server.
  - Moved configuration files into a `config` folder for better organization.
  - Setup database using `better-sqlite3` and added related database setup files.

- Authentication & User Management:
  - Implemented user registration and login endpoints.
  - Added OTP verification with time-to-live (TTL) for better security.
  - Introduced JWT signing and verification utilities.
  - Added features for resetting passwords and changing user passwords with re-authentication.
  - Validated email domains before registration.

- Database Enhancements:
  - Added necessary migrations and updated queries for user and authentication tables.
  - Implemented database triggers and fixed foreign key references.
  - Created async wrappers for controller functions and added utility functions for type assertions.

- Documentation:
  - Improved and fixed typos in documentation, especially related to response DTO.
  - Added API documentation for authentication routes.
  - Updated setup documentation for ease of use.

- Miscellaneous:
  - Refined logging for better error tracing.
  - Updated `.gitignore` to ignore variant `.env` files and unnecessary directories.

### CODE REVIEW

**Review by chimdie — CHANGES_REQUESTED** [review]
1. Update env sample on the Readme
2.  Update `/auth/register` reponse
Screenshot 2024-11-18 at 2 38 00 PM

**Comment by sammy-code98** [comment]
### Issue running setting up and running the server on my computer

Setting up and running the server on my local PC has proven abortive. I have attached a screenshot of the error below.

npm

The error occurs when I try to run `npm install`

it is worth noting that isolating or removing the `better-sqlite3` package from the project's dependencies means that  `npm install` runs successfully without errors

### PC /Environment specs
- Zorin/Ubuntu OS
- Node Version: v22.11.0
-Npm version: 10.9.0

**Comment by micaiah-effiong** [comment]
> ### Issue running setting up and running the server on my computer
> Setting up and running the server on my local PC has proven abortive. I have attached a screenshot of the error below.
> 
> npm
> 
> The error occurs when I try to run `npm install`
> 
> it is worth noting that isolating or removing the `better-sqlite3` package from the project's dependencies means that `npm install` runs successfully without errors
> 
> ### PC /Environment specs
> * Zorin/Ubuntu OS
> * Node Version: v22.11.0
> * Npm version: 10.9.0

What version of Ubuntu are you using?

File: server/src/docs/schemas.docs.ts
Comment (sammy-code98) [inline server/src/docs/schemas.docs.ts:1]: Noticed that this file is empty

**Review by sammy-code98 — APPROVED** [review]
LGTM except for the issue with me running it localluy

**Comment by sammy-code98** [comment]
### Issue running setting up and running the server on my computer

This issue has been resolved

### Cause of issue
A  system library `libc.so.6` required to run the `better-sqlite3` package was missing in the OS environment. The  `GLIBC` version `2.28`  available didnt match what was required to run the package `2.29`.

ps:  upgrading the zorin os to a version that comes with the right versions proved abortive as zorin doesnt support os upgrade from the terminal

### Solution
Use another os that has the `libc.so.6` and `GLIBC 2.29` installed.


### Server now runs well

goodnpm


Screenshot from 2024-11-21 09-14-11

**Pull request merged**